src/watchdog/utils/patterns.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# patterns.py: Common wildcard searching/filtering functionality for files.
+#
+# Copyright (C) 2010 Yesudeep Mangalapilly <yesudeep@gmail.com>
+#
+# Written by Boris Staletic <boris.staletic@gmail.com>
+
+from pathlib import PureWindowsPath, PurePosixPath
+
+
+def _match_path(path, included_patterns, excluded_patterns):
+    """Internal function same as :func:`match_path` but does not check arguments."""
+    common_patterns = included_patterns & excluded_patterns
+    if common_patterns:
+        raise ValueError(f'conflicting patterns `{common_patterns}` included and excluded')
```

Comment:
f-strings are available on Python 3.6+ only. But we are still supporting 3.4 and 3.5.